phpBB/phpbb/search/backend/base.php

```diff
@@ -400,10 +399,10 @@ public function delete_index(int &$post_counter = null): ?array
 				$this->index_remove($ids, $posters, $forum_ids);
 			}
 
-			$post_counter += self::BATCH_SIZE;
+			$post_counter = $ids[count($ids) - 1];
```

Comment:
You need to move this into the if now.

Reply:
Ok.
Should I also move `$row_count += count($ids);` to `if(...)` ?

Reply:
you don't have to, the point here is if `$ids` is empty, `count($ids) - 1` is `-1`, and not a valid index.

Reply:
Ahh! ok.
will do this changes in few mins.